src/timepicker/timepicker.spec.ts

```diff
@@ -1207,7 +1207,198 @@ describe('ngb-timepicker', () => {
   });
 
+  describe('Custom steps', () => {
+    const config = new NgbTimepickerConfig();
+    config.seconds = true;
+    config.hourStep = 2;
+    config.minuteStep = 3;
+    config.secondStep = 4;
+
+    beforeEach(() => {
+      TestBed.configureTestingModule(
+        {imports: [NgbTimepickerModule], providers: [{provide: NgbTimepickerConfig, useValue: config}]});
+    });
+
+    it('should increment / decrement hours by 6', async(() => {
+      const html = `<ngb-timepicker [(ngModel)]="model" [hourStep]="6"></ngb-timepicker>`;
+
+      const fixture = createTestComponent(html);
+      fixture.componentInstance.model = {hour: 10, minute: 30, second: 0};
+      fixture.detectChanges();
+      fixture.whenStable()
+        .then(() => {
+          fixture.detectChanges();
+          return fixture.whenStable();
+        })
+        .then(() => {
+
+          const buttons = getButtons(fixture.nativeElement);
+
+          expectToDisplayTime(fixture.nativeElement, '10:30:00');
+          expect(fixture.componentInstance.model).toEqual({hour: 10, minute: 30, second: 0});
+
+          (<HTMLButtonElement>buttons[0]).click();  // H+
+          fixture.detectChanges();
+          expectToDisplayTime(fixture.nativeElement, '16:30:00');
+          expect(fixture.componentInstance.model).toEqual({hour: 16, minute: 30, second: 0});
+
+          (<HTMLButtonElement>buttons[1]).click();  // H-
+          fixture.detectChanges();
+          expectToDisplayTime(fixture.nativeElement, '10:30:00');
+          expect(fixture.componentInstance.model).toEqual({hour: 10, minute: 30, second: 0});
+        });
+    }));
 
+    it('should increment / decrement hours to default value if step set to undefined', async(() => {
+      const html = `<ngb-timepicker [(ngModel)]="model" [hourStep]="undefined"></ngb-timepicker>`;
+
+      const fixture = createTestComponent(html);
+      fixture.componentInstance.model = {hour: 10, minute: 30, second: 0};
+      fixture.detectChanges();
+      fixture.whenStable()
+        .then(() => {
+          fixture.detectChanges();
+          return fixture.whenStable();
+        })
+        .then(() => {
+
+          const buttons = getButtons(fixture.nativeElement);
+
+          expectToDisplayTime(fixture.nativeElement, '10:30:00');
+          expect(fixture.componentInstance.model).toEqual({hour: 10, minute: 30, second: 0});
+
+          (<HTMLButtonElement>buttons[0]).click();  // H+
+          fixture.detectChanges();
+          expectToDisplayTime(fixture.nativeElement, '12:30:00');
+          expect(fixture.componentInstance.model).toEqual({hour: 12, minute: 30, second: 0});
+
+          (<HTMLButtonElement>buttons[1]).click();  // H-
+          fixture.detectChanges();
+          expectToDisplayTime(fixture.nativeElement, '10:30:00');
+          expect(fixture.componentInstance.model).toEqual({hour: 10, minute: 30, second: 0});
+        });
+    }));
+
+    it('should increment / decrement minutes by 7', async(() => {
```

Comment:
Could you please simplify tests, as we should use `async/await` or `fakeAsync` (I know you were inspired by already exsiting ones, but now there is a shorter way to write) → 

Before:

```ts
it('should increment / decrement minutes by 7', async(() => {
  //...
  fixture.detectChanges();
  fixture.whenStable()
    .then(() => {
      fixture.detectChanges();
      return fixture.whenStable();
    })
    .then(() => {
      // ...
    });
}));
```

After:
```ts
it('should increment / decrement minutes by 7', async(async () => {
  // ...
  fixture.detectChanges();
  await fixture.whenStable();
  fixture.detectChanges();
  // ...
}));
```